King County's Lost, Found, and Adoptable Pets: An Introduction

This is a short exploration of the open dataset provided by King County titled, "Lost, found, adoptable pets". I picked this data because I have a parrot, and the facebook groups I'm in frequently make posts about lost birds. This led me to wonder if there was an open dataset for lost animals. A simple google search led me to King County's open dataset repository, where I found the pet section (https://data.kingcounty.gov)

My goal with the data is to look at whether or not there is a difference in the proportions of dogs and cats that are considered "adoptable".

First, I wanted to gain an understanding of the type of data found within the dataset, so I looked at the beginning and end of the data

In [1]:
import pandas as pd
lfa = pd.read_csv('LostFoundAdoptable.csv')

lfa.head() 

impound_no Animal_ID                              Data_Source Record_Type  \
0  K22-145534   A662467  Regional Animal Services of King County        LOST   
1  K22-146235   A664212  Regional Animal Services of King County        LOST   
2  K22-145518   A662424  Regional Animal Services of King County        LOST   
3  K22-146310   A664391  Regional Animal Services of King County       FOUND   
4  K22-145976   A663552  Regional Animal Services of King County        LOST   

                                                Link      Current_Location  \
0  https://petharbor.com/PublicDetail.asp?searcht...                  LOST   
1  https://petharbor.com/PublicDetail.asp?searcht...                  LOST   
2  https://petharbor.com/PublicDetail.asp?searcht...                  LOST   
3  https://kingcounty.gov/depts/regional-animal-s...  In RASKC Foster Home   
4  https://petharbor.com/PublicDetail.asp?searcht...                  LOST   

  Animal_Name animal_type              Age Animal_Gender  ...     City State  \
0        Lola         Cat  Over 1 year<p/>        Female  ...      NaN   NaN   
1        Nino         Cat  Over 1 year<p/>          Male  ...      NaN   NaN   
2         Hua         Cat  Over 1 year<p/>          Male  ...  REDMOND    WA   
3         NaN         Cat          4 WEEKS          Male  ...   AUBURN    WA   
4       Maeby         Dog  Over 1 year<p/>        Female  ...      NaN   NaN   

       Zip  jurisdiction obfuscated_latitude obfuscated_longitude  \
0      NaN  JURISDICTION                 NaN                  NaN   
1      NaN  JURISDICTION                 NaN                  NaN   
2  98052.0       REDMOND                 NaN                  NaN   
3  98092.0   KING COUNTY                 NaN                  NaN   
4      NaN  JURISDICTION                 NaN                  NaN   

                                               Image  \
0  https://petharbor.com/get_image.asp?RES=Detail...   
1  https://petharbor.com/get_image.asp?RES=Detail...   
2  https://petharbor.com/get_image.asp?RES=Detail...   
3  https://petharbor.com/get_image.asp?RES=Detail...   
4  https://petharbor.com/get_image.asp?RES=Detail...   

                  image_alt_text  \
0  Image Copyright HLP Inc. 2022   
1  Image Copyright HLP Inc. 2022   
2  Image Copyright HLP Inc. 2022   
3  Image Copyright HLP Inc. 2022   
4  Image Copyright HLP Inc. 2022   

                                                Memo  Temperament  
0  Description: Lola is big and fluffy and loves ...          NaN  
1  Description: black/grey tabby. Small cyst unde...          NaN  
2  Description:  I Lost my CAT at Redmond Town Ce...          NaN  
3  Received on: 2022-05-18<p/> Found Near: 22400 ...          NaN  
4  Description: Black and Tan wire hair doxie abo...          NaN  

[5 rows x 25 columns]

In [2]:
lfa.tail()

impound_no Animal_ID                              Data_Source  \
397  K22-145752   A662969  Regional Animal Services of King County   
398  K22-145498   A662344  Regional Animal Services of King County   
399  K22-146397   A664593  Regional Animal Services of King County   
400  K22-146211   A664143  Regional Animal Services of King County   
401  K22-145948   A663493  Regional Animal Services of King County   

    Record_Type                                               Link  \
397       FOUND  https://petharbor.com/PublicDetail.asp?searcht...   
398        LOST  https://petharbor.com/PublicDetail.asp?searcht...   
399       FOUND    https://petharbor.com/pet.asp?uaid=KING.A664593   
400       FOUND  https://kingcounty.gov/depts/regional-animal-s...   
401       FOUND  https://petharbor.com/PublicDetail.asp?searcht...   

                                      Current_Location   Animal_Name  \
397                                     In Public Home  Kevin Caucho   
398                                               LOST          Gwen   
399  King County Pet Adoption Center 21615 64TH AVE...           NaN   
400                               In RASKC Foster Home  Fire Cracker   
401                                     In Public Home           NaN   

    animal_type              Age Animal_Gender  ...     City State      Zip  \
397         Dog      Unknown<p/>          Male  ...      NaN   NaN      NaN   
398         Cat  Over 1 year<p/>       Unknown  ...      NaN   NaN      NaN   
399         Cat          2 YEARS          Male  ...  SEATTLE    WA  98146.0   
400         Cat          7 WEEKS        Female  ...  KENMORE    WA  98028.0   
401         Cat      Unknown<p/>        Female  ...      NaN   NaN      NaN   

     jurisdiction obfuscated_latitude obfuscated_longitude  \
397  JURISDICTION                 NaN                  NaN   
398  JURISDICTION                 NaN                  NaN   
399   KING COUNTY                 NaN                  NaN   
400       KENMORE                 NaN                  NaN   
401  JURISDICTION                 NaN                  NaN   

                                                 Image  \
397  https://petharbor.com/get_image.asp?RES=Detail...   
398  https://petharbor.com/get_image.asp?RES=Detail...   
399  https://petharbor.com/get_image.asp?RES=Detail...   
400  https://petharbor.com/get_image.asp?RES=Detail...   
401  https://petharbor.com/get_image.asp?RES=Detail...   

                    image_alt_text  \
397  Image Copyright HLP Inc. 2022   
398  Image Copyright HLP Inc. 2022   
399  Image Copyright HLP Inc. 2022   
400  Image Copyright HLP Inc. 2022   
401  Image Copyright HLP Inc. 2022   

                                                  Memo  Temperament  
397  Description: Found this dog of 29th st se and ...          NaN  
398  Description: No collar, no chip, half bengal f...          NaN  
399  Received on: 2022-05-21<p/> Found Near: 11000 ...          NaN  
400  Received on: 2022-05-15<p/> Found Near: 7500 b...          NaN  
401  <p/>Age: Unknown<p/>Found Near: Kent West Hill...          NaN  

[5 rows x 25 columns]

Next, I wanted to obtain a greater undestanding of the total shape of the data:

In [4]:
lfa.shape

(402, 25)

This indicates that there are 402 rows and 25 columns. Upon physically reviewing the data, I found that there were different names for the same animal: Example: Cat vs. Barn Cat. Therefore, I needed to make sure I subsetted the data to contain the animal, and not a specific name or identifier. I used the following to subset all the data based on Animal Type, Gender, and Record Type (lost, found, or adoptable):

In [6]:
PetData = (lfa[['animal_type', 'Animal_Gender', 'Record_Type']])
print(PetData)

    animal_type Animal_Gender Record_Type
0           Cat        Female        LOST
1           Cat          Male        LOST
2           Cat          Male        LOST
3           Cat          Male       FOUND
4           Dog        Female        LOST
..          ...           ...         ...
397         Dog          Male       FOUND
398         Cat       Unknown        LOST
399         Cat          Male       FOUND
400         Cat        Female       FOUND
401         Cat        Female       FOUND

[402 rows x 3 columns]


I then performed the same function but with cats and dogs

In [7]:
CatData = PetData[PetData['animal_type'].str.contains('Cat')]
print(CatData)

    animal_type Animal_Gender Record_Type
0           Cat        Female        LOST
1           Cat          Male        LOST
2           Cat          Male        LOST
3           Cat          Male       FOUND
5           Cat          Male        LOST
..          ...           ...         ...
396    Dead Cat       Unknown       FOUND
398         Cat       Unknown        LOST
399         Cat          Male       FOUND
400         Cat        Female       FOUND
401         Cat        Female       FOUND

[247 rows x 3 columns]


In [8]:
DogData = PetData[PetData['animal_type'].str.contains('Dog')]
print(DogData)

    animal_type  Animal_Gender Record_Type
4           Dog         Female        LOST
8           Dog           Male        LOST
13          Dog           Male        LOST
15          Dog  Neutered Male   ADOPTABLE
18          Dog  Spayed Female       FOUND
..          ...            ...         ...
389         Dog  Neutered Male   ADOPTABLE
391         Dog        Unknown        LOST
393         Dog           Male       FOUND
395         Dog         Female       FOUND
397         Dog           Male       FOUND

[146 rows x 3 columns]


I then wanted to find out how many dogs and cats were considered to be adoptable, so I subsetted these data further:

In [16]:
AdoptableCat = CatData[CatData['Record_Type'] == 'ADOPTABLE']
print(AdoptableCat)

    animal_type  Animal_Gender Record_Type
38          Cat  Spayed Female   ADOPTABLE
58          Cat  Neutered Male   ADOPTABLE
62     Barn Cat  Neutered Male   ADOPTABLE
115         Cat  Neutered Male   ADOPTABLE
117         Cat  Neutered Male   ADOPTABLE
122         Cat  Spayed Female   ADOPTABLE
125         Cat  Spayed Female   ADOPTABLE
160         Cat  Spayed Female   ADOPTABLE
176         Cat  Spayed Female   ADOPTABLE
188         Cat  Neutered Male   ADOPTABLE
197         Cat  Spayed Female   ADOPTABLE
219         Cat  Spayed Female   ADOPTABLE
310         Cat  Neutered Male   ADOPTABLE
364         Cat  Neutered Male   ADOPTABLE
376         Cat  Neutered Male   ADOPTABLE


In [17]:
AdoptableDog = DogData[DogData['Record_Type'] == 'ADOPTABLE']
print(AdoptableDog)

    animal_type  Animal_Gender Record_Type
15          Dog  Neutered Male   ADOPTABLE
32          Dog  Neutered Male   ADOPTABLE
35          Dog  Neutered Male   ADOPTABLE
47          Dog  Spayed Female   ADOPTABLE
50          Dog  Spayed Female   ADOPTABLE
61          Dog  Neutered Male   ADOPTABLE
65          Dog  Neutered Male   ADOPTABLE
85          Dog  Neutered Male   ADOPTABLE
152         Dog  Neutered Male   ADOPTABLE
173         Dog  Neutered Male   ADOPTABLE
215         Dog  Neutered Male   ADOPTABLE
225         Dog  Neutered Male   ADOPTABLE
252         Dog  Neutered Male   ADOPTABLE
259         Dog  Neutered Male   ADOPTABLE
341         Dog  Neutered Male   ADOPTABLE
343         Dog  Spayed Female   ADOPTABLE
355         Dog  Neutered Male   ADOPTABLE
365         Dog  Spayed Female   ADOPTABLE
371         Dog  Neutered Male   ADOPTABLE
385         Dog  Neutered Male   ADOPTABLE
389         Dog  Neutered Male   ADOPTABLE


After reviewing the data, I developed the following question: at a alpha = 0.05 significance level, does sufficient evidence exist showing that the proportion of adoptable dogs is greater than that of adoptable cats? This creates the following hypotheses: 
Null hypothesis: There is no significant difference in the proportion of adoptability between dogs and cats
Alternate hypothesis: There is a significant difference in the proportion of adoptable dogs than adoptable cats.
Therefore, our analysis will be right tailed

In [21]:
from statsmodels.stats.proportion import proportions_ztest
adopt = [len(AdoptableCat), len(AdoptableDog)]
n = [len(DogData), len(CatData)]
print(proportions_ztest(adopt, n))

(0.5884192476325393, 0.5562509215341186)


The proportions_ztest indicates a z-score of 0.588, with a p-value of 0.57. Since the p-value 0.57 > 0.05, we can infer that there is no significant difference in the proportion of adoptability between cats and dogs.